dp_mobility_report/model/od_analysis.py

```diff
@@ -13,32 +13,56 @@
 from dp_mobility_report.privacy import diff_privacy
 
 
-def get_od_shape(df: pd.DataFrame, tessellation: GeoDataFrame) -> pd.DataFrame:
-    ends_od_shape = (
-        df[
-            (df[const.POINT_TYPE] == const.END)
-            & df[const.TILE_ID].isin(tessellation[const.TILE_ID])
-        ][[const.TID, const.TILE_ID, const.DATETIME, const.LAT, const.LNG]]
-        .merge(tessellation[[const.TILE_ID]], on=const.TILE_ID, how="left")
-        .rename(
-            columns={
-                const.TILE_ID: const.TILE_ID_END,
-                const.LAT: const.LAT_END,
-                const.LNG: const.LNG_END,
-                const.DATETIME: const.DATETIME_END,
-            }
+def get_od_shape(df: pd.DataFrame, tessellation: GeoDataFrame, timestamps: bool=True) -> pd.DataFrame:
+    if timestamps:
+        ends_od_shape = (
```

Comment:
maybe a little more elegant without code duplicate (did not test the code yet!):

```python
if timestamps:
    cols = [const.TID, const.TILE_ID, const.DATETIME, const.LAT, const.LNG]
else:
    cols = [const.TID, const.TILE_ID, const.LAT, const.LNG]

ends_od_shape = (
            (df[const.POINT_TYPE] == const.END)
            df[
            & df[const.TILE_ID].isin(tessellation[const.TILE_ID])
                (df[const.POINT_TYPE] == const.END)
        ][cols]
                & df[const.TILE_ID].isin(tessellation[const.TILE_ID])
        .merge(tessellation[[const.TILE_ID]], on=const.TILE_ID, how="left")
            ][cols]
        .rename(
            .merge(tessellation[[const.TILE_ID]], on=const.TILE_ID, how="left")
            columns={
            .rename(
                const.TILE_ID: const.TILE_ID_END,
                columns={
                const.LAT: const.LAT_END,
                    const.TILE_ID: const.TILE_ID_END,
                const.LNG: const.LNG_END,
                    const.LAT: const.LAT_END,
                    const.LNG: const.LNG_END,
            }

if timestamps:
  ends_od_shape = ends_od_shape.rename({
                      const.DATETIME: const.DATETIME_END
                  }
              )

od_shape = (
            df[
                (df[const.POINT_TYPE] == const.START)
                & df[const.TILE_ID].isin(tessellation[const.TILE_ID])
            ][cols]
            .merge(tessellation[[const.TILE_ID]], on=const.TILE_ID, how="left")
            .merge(ends_od_shape, on=const.TID, how="inner")
````